pkg/util/cgroups/pid_mapper.go

```diff
@@ -165,12 +166,12 @@ func (pm *procPidMapper) refreshMapping(cacheValidity time.Duration) {
 				return godirwalk.SkipThis
 			}
 
-			containerID, cgroupRelativePath, err := ContainerIDFromCgroupReferences(pm.procPath, de.Name(), pm.cgroupController)
+			cgroupIdentifier, err := IdentiferFromCgroupReferences(pm.procPath, de.Name(), pm.cgroupController, pm.readerFilter)
 			if err != nil {
 				log.Debugf("Unable to parse cgroup file for pid: %s, err: %v", de.Name(), err)
 			}
-			if containerID != "" {
-				cgroupPidMapping[cgroupRelativePath] = append(cgroupPidMapping[cgroupRelativePath], int(pid))
+			if cgroupIdentifier != "" {
+				cgroupPidMapping[cgroupIdentifier] = append(cgroupPidMapping[cgroupIdentifier], int(pid))
```

Comment:
based on the "code scanning"

```suggestion
				cgroupPidMapping[cgroupIdentifier] = append(cgroupPidMapping[cgroupIdentifier], int64(pid))
```